test/test_helpers.jl

```diff
@@ -3,16 +3,63 @@ struct Multiplier{T}
     x::T
 end
 (m::Multiplier)(y) = m.x * y
+(m::Multiplier)(y, z) = (m.x * y) * z
+
 function ChainRulesCore.rrule(m::Multiplier, y)
-    Multiplier_pullback(z̄) = Tangent{typeof(m)}(; x=y * z̄), m.x * z̄
+    Multiplier_pullback(dΩ) = (Tangent{typeof(m)}(; x = dΩ * y'), m.x' * dΩ)
     return m(y), Multiplier_pullback
 end
+function ChainRulesCore.rrule(m::Multiplier, y, z)
+    Multiplier_pullback_3(dΩ) = (
+        Tangent{typeof(m)}(; x = dΩ * (y * z)'),
+        m.x' * dΩ * z', 
+        (m.x * y)' * dΩ,
+    )
+    return m(y, z), Multiplier_pullback_3
+end
+
+# A functor which counts up!
+mutable struct Counter
```

Comment:
Needs a docstring saying how to use this, since it is doing weird gradiet things